crates/iceberg/src/spec/schema.rs

```diff
@@ -609,13 +611,88 @@ impl SchemaVisitor for IndexByName {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+#[serde(rename_all = "kebab-case")]
+/// Names and types of fields in a table.
+pub(crate) struct SchemaV2 {
+    /// Identifier of the schema
+    pub schema_id: i32,
+    /// Set of primitive fields that identify rows in a table.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub identifier_field_ids: Option<Vec<i32>>,
+
+    #[serde(flatten)]
+    /// The struct fields
+    pub fields: StructType,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+#[serde(rename_all = "kebab-case")]
+/// Names and types of fields in a table.
+pub(crate) struct SchemaV1 {
```

Comment:
Do we really need the `SchemaV1`. What we do in PyIceberg is when reading a V1, and when encounter a Schema in the `schema` field:

![image](https://github.com/apache/iceberg-rust/assets/1134248/74e3e5fa-4e8b-4dff-90e7-4f0a0a112c67)

We just set the schema-id to 0, and add the schema to the schema's field. We consider the metadata equivalent.

Reply:
`SchemaV1` (and `SchemaV2`) are internal structs and are not visible to a user of the library. `SchemaV1` is just used for serialization/deserialization. We can still do your recommended conversion.